samples/cpp/qrcode.cpp

```diff
@@ -157,7 +160,7 @@ void drawQRCodeResults(Mat& frame, const vector<Point>& corners, const vector<cv
 
 static
 void runQR(
-    QRCodeDetector& qrcode, const Mat& input,
+    Ptr<QRCodeDetectorBase> qrcode, const Mat& input,
```

Comment:
1. We don't need `Ptr` for PImpl which is already smart ptr.
2. `Ptr`s must be passed as "const reference" be default.

Reply:
removed Ptr, added ref